### ISSUE TITLE
Incorrect syntax in for loop crashes RF and html reports are not generated.

### ISSUE THREAD

**Issue by rasjani** [robotframework/robotframework#5120]
With "wrong syntax" for FOR loop:

```robotframework
  FOR     ${button_id}    in    @{DEFAULT_PLANS}
    Sleep  1 seconds
  END
```
where `in` ofcourse should be `IN` - causes robot framework to crash without generating log and report html files.

Currently installed packages:
```
click==8.1.7
grpcio==1.62.1
grpcio-tools==1.62.1
overrides==7.7.0
protobuf==4.25.3
robotframework==7.0
robotframework-assertion-engine==3.0.3
robotframework-browser==18.3.0
robotframework-pythonlibcore==4.4.1
setuptools==69.5.1
typing_extensions==4.11.0
wheel==0.43.0
wrapt==1.16.0
```

Python version: 3.12.2

Stacktrace:

```
[ ERROR ] Unexpected error: TypeError: sequence item 4: expected str instance, NoneType found
Traceback (most recent call last):
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/utils/application.py", line 81, in _execute
    rc = self.main(arguments, **options)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/run.py", line 460, in main
    result = suite.run(settings)
             ^^^^^^^^^^^^^^^^^^^
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/running/model.py", line 758, in run
    self.visit(runner)
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/model/testsuite.py", line 420, in visit
    visitor.visit_suite(self)
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/model/visitor.py", line 132, in visit_suite
    suite.tests.visit(self)
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/model/itemlist.py", line 102, in visit
    item.visit(visitor)    # type: ignore
    ^^^^^^^^^^^^^^^^^^^
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/model/testcase.py", line 177, in visit
    visitor.visit_test(self)
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/running/suiterunner.py", line 169, in visit_test
    runner.run(data, result)
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/running/bodyrunner.py", line 49, in run
    item.run(result, self._context, self._run, self._templated)
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/running/model.py", line 196, in run
    return ForRunner(context, self.flavor, run, templated).run(self, result)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/running/bodyrunner.py", line 102, in run
    with StatusReporter(data, result, self._context, run) as status:
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/running/statusreporter.py", line 45, in __enter__
    context.start_body_item(self.data, result, self.implementation)
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/running/context.py", line 291, in start_body_item
    method(*args)
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/output/output.py", line 97, in start_for
    LOGGER.start_for(data, result)
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/output/logger.py", line 32, in wrapper
    method(self, *args)
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/output/logger.py", line 274, in start_for
    logger.start_for(data, result)
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/output/listeners.py", line 133, in start_for
    listener.start_for(data, result)
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/output/listeners.py", line 493, in start_for
    self._start_kw(result._log_name, self._attrs(data, result, **extra))
                   ^^^^^^^^^^^^^^^^
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/result/model.py", line 348, in _log_name
    return str(self)[7:]    # Drop 'FOR    ' prefix.
           ^^^^^^^^^
  File "/Users/rasjani/src/work/clientname/projectname/venv/lib/python3.12/site-packages/robot/model/control.py", line 148, in __str__
    return '    '.join(parts)
           ^^^^^^^^^^^^^^^^^^
TypeError: sequence item 4: expected str instance, NoneType found
```

I'd expect to get some sort of error ofcourse but not a full blown crash.

**Comment by pekkaklarck** [comment]
Thanks for the report. This seems to a regression in RF 7.0, because the example works fine with RF 6.1. Ought to be easy to fix but we absolutely need to add tests for this as well to avoid it occurring again. Kind of good that RF 7.0.1 has been so badly delayed, because we can still get this into that release.

**Comment by pekkaklarck** [comment]
We actually already had tests for invalid separators, but this bug occurred only when generating the log file and our tests don't typically generate them. I changed the relevant tests so that logs are generated and fixed the problem.

The problem was that because there was no valid separator used, the parser generated a `For` model object with `None` as a separator and that wasn't handled properly elsewhere. I decided the fix the issue so that the separator is a string `IN` in that case instead. Alternatively could have allowed it to be `None` and just take that into account elsewhere, but that would have also required changing typing from the current `Literal['IN', 'IN RANGE', ...] = 'IN'` to something that allows `None`.
